README.md

```diff
@@ -95,15 +95,56 @@ If the second query doesn't depend on the other one, then you can use GraphQL::B
 
 ```ruby
 resolve -> (obj, args, context) do
-  smart_collection_query = CountQuery.new(SmartCollection, context.shop_id)
-  custom_collection_query = CountQuery.new(CustomCollection, context.shop_id)
+  QueryGroup.new([smart_collection_query, custom_collection_query]).then do |results|
+    results.reduce(&:+)
+  end
+end
+
+### Error Handling
 
-  QueryGroup.new([smart_collection_query, custom_collection_query]) do
-    smart_collection_query.result + custom_collection_query.result
+Exceptions can be rescued by using `.rescue(error_class)` instead of `.then` which could be used for a fallback
+
+```ruby
+resolve -> (obj, args, context) do
+  CacheFetchQuery.new(Product, args["id"]).rescue(Redis::BaseConnectionError) do |err|
+    logger.warn err.message
+    FindQuery.new(Product, args["id"])
   end
 end
 ```
 
+If you want to do something after the query without affecting the result, then you can use `.ensure` which could be used for instrumentation
```

Comment:
I'm assuming `ensure` is run even if an exception occurs... might want to make that explicit